comment/conf/defaults.py

```diff
@@ -13,3 +14,8 @@
 COMMENT_URL_PREFIX = 'comment-'
 COMMENT_URL_SUFFIX = ''
 COMMENT_URL_ID_LENGTH = 8
+COMMENT_ALLOW_ANONYMOUS = False
+if getattr(settings, 'COMMENT_ALLOW_ANONYMOUS', COMMENT_ALLOW_ANONYMOUS):
+    COMMENT_FROM_EMAIL = settings.EMAIL_HOST_USER   # used for sending confirmation emails
+    COMMENT_CONTACT_EMAIL = COMMENT_FROM_EMAIL  # used for contact address in confirmation emails
```

Comment:
> > ![anonymous-comment](https://user-images.githubusercontent.com/13880786/90076049-6ca81900-dd1c-11ea-839e-441f40b363f6.gif)
> 
> The user shall be kept in the same view after posting the comment, redirecting to the home page is not user friendly smile

They aren't redirected. You replace the `comment` html container with the data received from the backend. Since I return a `JsonResponse` currently and I have not made any changes to `javascript`, it might seem like a redirect. Since the response is not handled anyway right now at the frontend, the container appears to be empty and hence a redirect.


> we also should look at improving the UI for anonymous case

Yes, any suggestions would be great.

Reply:
Okay, I'm a bit confused here. Do we define the default values inside this file or the actual values of these variables here.

As far as I see, we are defining the default values here. If that is the case, the default values of the variables `COMMENT_FROM_EMAIL` and `COMMENT_CONTACT_EMAIL` are same unless specified otherwise by the user in their settings.

Reply:
so in this case it should be like this:
```python
COMMENT_CONTACT_EMAIL = getattr(settings, 'COMMENT_CONTACT_EMAIL', COMMENT_FROM_EMAIL)
```

Reply:
Okay, this is what I thought.

Shouldn't the other variables defined inside this file like
```python

COMMENT_FLAG_REASONS = [
    (1, _('Spam | Exists only to promote a service')),
    (2, _('Abusive | Intended at promoting hatred')),
]
```
be defined the same way as:

```python

COMMENT_FLAG_REASONS = getattr(settings, 'COMMENT_FLAG_REASONS', [
    (1, _('Spam | Exists only to promote a service')),
    (2, _('Abusive | Intended at promoting hatred')),
])

```

Reply:
Hmmm, I got what you mean. The developer settings will override the value of any existing attribute in the default settings.

so I would prefer going this way:

```python
COMMENT_FLAG_REASONS = [
    (1, _('Spam | Exists only to promote a service')),
    (2, _('Abusive | Intended at promoting hatred')),
]
```

Reply:
Yes. If you see the GIF attached closely the contact email is `contact@domain`(picked from developer settings) which has overriden the default values
That means I don't have to make any changes as suggested earlier in this discussion?

Reply:
You are right. It's confusing. 😄